ambry-protocol/src/main/java/com.github.ambry.protocol/DeleteRequest.java

```diff
@@ -27,27 +28,45 @@
  */
 public class DeleteRequest extends RequestOrResponse {
   private BlobId blobId;
+  private short accountId;
+  private short containerId;
+  private int deletionTimeInSecs;
   private int sizeSent;
-  private static final short Delete_Request_Version_V1 = 1;
+  static final short Delete_Request_Version_V1 = 1;
+  static final short Delete_Request_Version_V2 = 2;
+  private static short currentVersion = Delete_Request_Version_V1;
 
+  private static final int AccountId_ContainerId_Field_Size_InBytes = 2;
+  private static final int DeletionTime_Field_Size_InBytes = 4;
+
+  // @TODO: remove this constructor once DeleteRequest V2 is enabled
   public DeleteRequest(int correlationId, String clientId, BlobId blobId) {
-    super(RequestOrResponseType.DeleteRequest, Delete_Request_Version_V1, correlationId, clientId);
+    super(RequestOrResponseType.DeleteRequest, currentVersion, correlationId, clientId);
     this.blobId = blobId;
```

Comment:
the other constructor will be used while enabling write path. As of now, it is used only in tests. This is to ensure, that we can test the read path for V2. Will sync up with you offline